cpp/daal/src/algorithms/k_nearest_neighbors/oneapi/bf_knn_classification_predict_kernel_ucapi_impl.i

```diff
@@ -114,8 +137,8 @@ services::Status KNNClassificationPredictKernelUCAPI<algorithmFpType>::compute(c
     // Block dimensions below are optimal for GEN9
     // Number of doubles is to 2X less against floats
     // to keep the same block size in bytes
-    const uint32_t maxDataBlockRowCount  = 4096 * 4;
-    const uint32_t maxQueryBlockRowCount = (2048 * 4) / sizeof(algorithmFpType);
+    const uint32_t maxDataBlockRowCount  = 4096;
```

Comment:
Have you checked that new block dimensions are more efficient?

Reply:
Not yet. I'll check it

Reply:
I've returned original values and also I'll try a few more options to maximize performance